src/v/kafka/server/tests/alter_config_test.cc

```diff
@@ -886,3 +885,171 @@ FIXTURE_TEST(test_alter_config_internal_topic, alter_config_test_fixture) {
+          std::make_pair("false", kafka::config_resource_operation::set));
+        properties.emplace(
+          "redpanda.remote.read",
+          std::make_pair("true", kafka::config_resource_operation::set));
+        test_cases.push_back(std::move(properties));
+    }
+    {
+        map_t properties;
+        properties.emplace(
+          "redpanda.remote.write",
+          std::make_pair("true", kafka::config_resource_operation::set));
+        properties.emplace(
+          "redpanda.remote.read",
+          std::make_pair("false", kafka::config_resource_operation::set));
+        test_cases.push_back(std::move(properties));
+    }
+    {
+        map_t properties;
+        properties.emplace(
+          "redpanda.remote.write",
+          std::make_pair("true", kafka::config_resource_operation::set));
+        properties.emplace(
+          "redpanda.remote.read",
+          std::make_pair("true", kafka::config_resource_operation::set));
+        test_cases.push_back(std::move(properties));
+    }
+    {
+        map_t properties;
+        properties.emplace(
+          "redpanda.remote.write",
+          std::make_pair("false", kafka::config_resource_operation::set));
+        properties.emplace(
+          "redpanda.remote.read",
+          std::make_pair("false", kafka::config_resource_operation::set));
+        test_cases.push_back(std::move(properties));
+    }
+    {
+        map_t properties;
+        properties.emplace(
+          "redpanda.remote.write",
+          std::make_pair(
+            std::nullopt, kafka::config_resource_operation::remove));
+        properties.emplace(
+          "redpanda.remote.read",
+          std::make_pair("true", kafka::config_resource_operation::set));
+        test_cases.push_back(std::move(properties));
+    }
+    {
+        map_t properties;
+        properties.emplace(
+          "redpanda.remote.write",
+          std::make_pair("true", kafka::config_resource_operation::set));
+        properties.emplace(
+          "redpanda.remote.read",
+          std::make_pair(
+            std::nullopt, kafka::config_resource_operation::remove));
+        test_cases.push_back(std::move(properties));
+    }
+    {
+        map_t properties;
+        properties.emplace(
+          "redpanda.remote.write",
+          std::make_pair("true", kafka::config_resource_operation::set));
+        properties.emplace(
+          "redpanda.remote.read",
+          std::make_pair("true", kafka::config_resource_operation::set));
+        test_cases.push_back(std::move(properties));
+    }
+    {
+        map_t properties;
+        properties.emplace(
+          "redpanda.remote.write",
+          std::make_pair(
+            std::nullopt, kafka::config_resource_operation::remove));
+        properties.emplace(
+          "redpanda.remote.read",
+          std::make_pair(
+            std::nullopt, kafka::config_resource_operation::remove));
+        test_cases.push_back(std::move(properties));
+    }
```

Comment:
Could you please add another test (either here or as a ducktape test) that verifies that remove now resets to the cluster-level default instead of resetting to false? I think we don't have that yet.

Reply:
We have this in `alter_topic_configuration_test.py`, `test_overrides_remove()`.

Added an extra test case here as well.